public/js/action-list.js

```diff
@@ -28,194 +28,622 @@
+            req.done(function () {
+                showMore.remove();
+
+                // Set data-icinga-url to make it available for Icinga.History.getCurrentState()
+                req.$target.closest('.container').data('icingaUrl', url);
+
+                this.icinga.history.replaceCurrentState();
+            });
+
+            return req;
+        }
+
+        /**
+         * Create the detail url for multi selectable list
+         *
+         * @param items List items
+         * @param withBaseUrl Default to true
+         *
+         * @returns {string} The url
+         */
+        createMultiSelectUrl(items, withBaseUrl = true) {
+            let filters = [];
+            items.forEach(item => {
+                filters.push(item.getAttribute('data-icinga-multiselect-filter'));
+            });
+
+            let url = '?' + filters.join('|');
+
+            if (withBaseUrl) {
+                return items[0].parentElement.getAttribute('data-icinga-multiselect-url') + url;
+            }
+
+            return url;
+        }
+
+        onColumnClose(event) {
+            if (event.target.id !== 'col2') {
+                return;
+            }
+
+            let list = document.querySelector('#col1 .action-list');
+            if (list && list.matches('[data-icinga-multiselect-url], [data-icinga-detail-url]')) {
+                let _this = event.data.self;
+                _this.clearSelection(list.querySelectorAll(':scope > [data-action-item].active'));
+                _this.addSelectionCountToFooter(list);
+            }
+        }
+
+        /**
+         * Triggers when column is moved to left or right
+         *
+         * @param event
+         * @param sourceId The content is moved from
+         */
+        onColumnMoved (event, sourceId) {
+            let _this = event.data.self;
+
+            if (event.target.id === 'col2' && sourceId === 'col1') { // only for browser-back (col1 shifted to col2)
+                _this.clearSelection(event.target.querySelectorAll('.action-list .active'));
             }
         }
 
         onRendered(event) {
+            let _this = event.data.self;
             let container = event.target;
             let isTopLevelContainer = container.matches('#main > :scope');
+            let detailUrl = _this.icinga.utils.parseUrl(
+                _this.icinga.history.getCol2State().replace(/^#!/, '')
+            );
+            let list = null;
 
-            if (event.currentTarget !== container) {
-                // Nested containers are not processed multiple times
+            if (event.currentTarget !== container || _this.lastTimeoutId !== null) {
+                // Nested containers are not processed multiple times || still processing selection/navigation
                 return;
             } else if (isTopLevelContainer && container.id !== 'col1') {
-                return;
-            }
+                list = document.querySelector('#main > #col1 .action-list');
```

Comment:
I believe the list can be identified by identifying the item based on `detailUrl` first. Then the list can be anywhere and we're not limited to `#col1`.